### ISSUE TITLE
Sumcheck loop ordering

### ISSUE THREAD

**Issue by lucasxia01** [AztecProtocol/barretenberg#728]
Brought up in a discussion with Zac over potential slowdowns in Sumcheck.

Currently we iterate over each polynomial and for each one, we call `add_scaled`, which loops over the same `batched_poly_unshifted` and `batched_poly_to_be_shifted` memory over and over again. Instead, we should consider switching the order of the loops so that we iterate over the polynomial indices in the outer loop and over the polynomials in the inner loop.

**Comment by ledwards2225** [comment]
@lucasxia01 Is this still relevant? If so, can you clarify what part of the code this refers to?

**Comment by lucasxia01** [comment]
I don't remember exactly what this is referring to. Should've attached it to a line in the code. Maybe @zac-williamson would know.

**Comment by ledwards2225** [comment]
I assume this is referring to the actual polynomial folding in PG. Zac has [this](https://github.com/AztecProtocol/aztec-packages/pull/11748) PR which could be picked up if we want to handle this.

**Comment by lucasxia01** [comment]
No, its related to sumcheck going row by row instead of column by column.

**Comment by ledwards2225** [comment]
I cant make any sense of how sumcheck could be executed column by column :)
